analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -497,9 +577,17 @@ class Pub(
         return yamlMapper.readTree(definitionFile)
     }
 
-    override fun command(workingDir: File?) = if (Os.isWindows) "pub.bat" else "pub"
+    override fun command(workingDir: File?): String {
+        var cmd = if (Os.isWindows) "pub.bat" else "pub"
+        if (pubAbsolutePath.isDirectory) cmd = "$pubAbsolutePath" + File.separator + cmd
+        return cmd
```

Comment:
Nit: Could make `cmd` a `val` and write

    return if (pubAbsolutePath.isDirectory) "$pubAbsolutePath${File.separator}$cmd" else cmd

Similar below.